server/src/main/java/org/opensearch/rest/action/admin/cluster/RestClusterRerouteAction.java

```diff
@@ -104,6 +111,14 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         final String metric = request.param("metric");
         if (metric == null) {
             request.params().put("metric", DEFAULT_METRICS);
+        } else {
+            // TODO: Remove the statements in 'else' after removing MASTER_ROLE.
+            EnumSet<ClusterState.Metric> metrics = ClusterState.Metric.parseString(request.param("metric"), true);
+            // Because "_all" value will add all Metric into metrics set, for prevent deprecation message shown in that case,
+            // add the check of validating metrics set doesn't contain all enum elements.
+            if (!metrics.equals(EnumSet.allOf(ClusterState.Metric.class)) && metrics.contains(ClusterState.Metric.MASTER_NODE)) {
+                deprecationLogger.deprecate("cluster_reroute_metric_parameter_master_node_value", DEPRECATED_MESSAGE_MASTER_NODE);
+            }
         }
         return channel -> client.admin().cluster().reroute(clusterRerouteRequest, new RestToXContentListener<>(channel));
```

Comment:
For my understanding - are there no changes needed here to support/parse the new `cluster_manager_node` parameter?

Reply:
The new `cluster_manager_node` is a value of the request parameter `metric`. I think the only code to support is on above, where you posted a comment 😄.
Since it worked, I haven't revise the whole process of the parameter parsing.